ydb/aio/query/transaction.py

```diff
@@ -0,0 +1,151 @@
+        """
+        Closes a transaction context manager and rollbacks transaction if
+        it is not finished explicitly
+        """
+        self._ensure_prev_stream_finished()
+        if self._tx_state._state == QueryTxStateEnum.BEGINED:
+            # It's strictly recommended to close transactions directly
+            # by using commit_tx=True flag while executing statement or by
+            # .commit() or .rollback() methods, but here we trying to do best
+            # effort to avoid useless open transactions
+            logger.warning("Potentially leaked tx: %s", self._tx_state.tx_id)
+            try:
+                await self.rollback()
+            except issues.Error:
+                logger.warning("Failed to rollback leaked tx: %s", self._tx_state.tx_id)
+
+    async def _ensure_prev_stream_finished(self) -> None:
+        if self._prev_stream is not None:
+            async for _ in self._prev_stream:
+                pass
+            self._prev_stream = None
+
+    async def begin(self, settings: Optional[base.QueryClientSettings] = None) -> "QueryTxContextAsync":
+        """WARNING: This API is experimental and could be changed.
+
+        Explicitly begins a transaction
+
+        :param settings: A request settings
+
+        :return: None or exception if begin is failed
+        """
+        await self._begin_call(settings)
+
+    async def commit(self, settings: Optional[base.QueryClientSettings] = None) -> None:
+        """WARNING: This API is experimental and could be changed.
+
+        Calls commit on a transaction if it is open otherwise is no-op. If transaction execution
+        failed then this method raises PreconditionFailed.
+
+        :param settings: A request settings
+
+        :return: A committed transaction or exception if commit is failed
+        """
+        if self._tx_state._already_in(QueryTxStateEnum.COMMITTED):
+            return
+
+        if self._tx_state._state == QueryTxStateEnum.NOT_INITIALIZED:
+            self._tx_state._change_state(QueryTxStateEnum.COMMITTED)
+            return
+
+        await self._ensure_prev_stream_finished()
+
+        await self._commit_call(settings)
+
+    async def rollback(self, settings: Optional[base.QueryClientSettings] = None) -> None:
+        """WARNING: This API is experimental and could be changed.
+
+        Calls rollback on a transaction if it is open otherwise is no-op. If transaction execution
+        failed then this method raises PreconditionFailed.
+
+        :param settings: A request settings
+
+        :return: A committed transaction or exception if commit is failed
+        """
+        if self._tx_state._already_in(QueryTxStateEnum.ROLLBACKED):
+            return
+
+        if self._tx_state._state == QueryTxStateEnum.NOT_INITIALIZED:
+            self._tx_state._change_state(QueryTxStateEnum.ROLLBACKED)
+            return
+
+        await self._ensure_prev_stream_finished()
```

Comment:
The method override base method, but with other signature.
It is difference from sync class: sync class use method from base class.

What about do one of:
1. Extract common part to base class and reuse it from both children
2. Complete split method to sync and async - for same style.

Now I can fix bug in the base class and think about I fix the bug/change behaviour in both variants because it is base class. (Or forgot about async code at all).

Same for other methods from the base class.